src/components/ViewSnippetSidebar.tsx

```diff
@@ -23,10 +23,27 @@ import { useFilesDialog } from "./dialogs/files-dialog"
 
 export default function ViewSnippetSidebar({
   className,
-}: { className?: string }) {
-  const { snippet } = useCurrentSnippet()
-  const { toast } = useToast()
-  const { Dialog: FilesDialog, openDialog: openFilesDialog } = useFilesDialog()
+}: {
+  className?: string;
+}) {
+  const { snippet } = useCurrentSnippet();
+  const { toast } = useToast();
+  const { Dialog: FilesDialog, openDialog: openFilesDialog } = useFilesDialog();
+
+  const copyToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      toast({
+        title: "Copied to clipboard!",
+        description: "The text has been copied successfully.",
+      });
+    } catch (error) {
+      toast({
+        title: "Failed to copy",
+        description: "There was an error copying the text.",
+      });
+    }
+  };
```

Comment:
can you put this in a separate hook called `useCopyToClipboard`

Also you'll need to run `bun format` to match the code style

Reply:
Fixed